daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala

```diff
@@ -181,20 +198,30 @@ private final class MatchingStringModule(description: String, string_regex: Stri
   */
 private final class ConcatenableMatchingStringModule(
     description: String,
-    extraAllowedChars: Char => Boolean,
+    extraAllowedChars: String,
     maxLength: Int = Int.MaxValue,
 ) extends StringModuleImpl
     with ConcatenableStringModule[String, String] {
 
-  override def fromString(s: String): Either[String, T] =
-    if (s.isEmpty)
-      Left(s"""$description is empty""")
-    else if (s.length > maxLength)
-      Left(s"""$description is too long""")
-    else
-      s.find(c => c > 0x7f || !(c.isLetterOrDigit || extraAllowedChars(c)))
-        .fold[Either[String, T]](Right(s))(c =>
-          Left(s"""non expected character 0x${c.toInt.toHexString} in $description "$s""""))
+  private val disallowedChar =
+    IdString.asciiCharsToRejectionArray(IdString.letters ++ IdString.digits ++ extraAllowedChars)
```

Comment:
How often do we call this? I assume only once during the runtime of the participant node. Is that correct?

Reply:
This will be called once for each instance of `ConcatenableStringModule` (we have 4 of them) and twice for the object `Name` .
That is a total of 6 times for a participant node.